indexer/pkg/worker/heap.go

```diff
@@ -0,0 +1,83 @@
+package worker
+
+import (
+	"container/heap"
+	"time"
+)
+
+// DelayHeap implements a min-heap for scheduling delayed task execution.
+// It orders tasks by their runAt time, with the earliest scheduled task
+// at the root. This allows the worker pool to efficiently manage retries
+// by enqueuing tasks that aren't ready for immediate execution.
+//
+// DelayHeap implements the heap.Interface from the container/heap package
+// and can be used with the standard heap operations:
+//
+//	h := &DelayHeap{}
+//	heap.Init(h)
+//	heap.Push(h, task)
+//	task := heap.Pop(h).(*Task)
+//
+// The heap maintains the index field of each Task to support efficient
+// heap operations and reordering.
+type DelayHeap []*Task
```

Comment:
Not for now necessarily but I am wondering how similar this heap is to the executor one. Could we somehow template it and reuse it across 🤔